Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,240 @@
+    /// Important thing to keep in mind that only properties present in `options` will be updated,
+    /// all other will remain the same as specified before.
+    ///
+    /// - Throws:
+    ///   - `ViewAnnotationManagerError.annotationNotFound`: the supplioed view was not found
+    ///   - `ViewAnnotationManagerError.updateFailure`: errors during the up date of the view (eg. incorrect parameters)
+    public func update(_ annotatonView: AnnotationView, options: ViewAnnotationOptions) throws {
+        guard let id = annotationIdByViews[annotatonView] else {
+            throw ViewAnnotationManagerError.annotationNotFound
+        }
+        try mapboxMap.updateViewAnnotation(withId: id, options: options)
+    }
+
+    /// Find `AnnotationView` by feature id if it was specified as part of `ViewAnnotationOptions.associatedFeatureId`.
+    ///
+    /// - Returns: `AnnotationView` if view was found and `nil` otherwise.
+    public func viewAnnotation(byFeatureId identifier: String) -> AnnotationView? {
+        guard let id = annotationViewsById.keys.first(where: { id in
+            (try? mapboxMap.options(forViewAnnotationWithId: id).associatedFeatureId == identifier) ?? false
+        }) else {
+            return nil
+        }
+        return annotationViewsById[id]
+    }
+
+    /// Find `ViewAnnotationOptions` of view annotation by feature id if it was specified as part of `ViewAnnotationOptions.associatedFeatureId`.
+    ///
+    /// - Returns: `ViewAnnotationOptions` if view was found and `nil` otherwise.
+    public func options(byFeatureId identifier: String) -> ViewAnnotationOptions? {
+        return viewAnnotation(byFeatureId: identifier).flatMap { options(byAnnotationView: $0) }
+    }
+
+    /// Get current `ViewAnnotationOptions` for given `AnnotationView`.
+    ///
+    /// - Returns: `ViewAnnotationOptions` if view was found and `nil` otherwise.
+    public func options(byAnnotationView view: AnnotationView) -> ViewAnnotationOptions? {
+        return annotationIdByViews[view].flatMap { try? mapboxMap.options(forViewAnnotationWithId: $0) }
+    }
+
+    // MARK: - Internal functions
+
+    internal func placeAnnotations(positions: [ViewAnnotationPositionDescriptor]) {
+        // Iterate through and update all view annotations
+        // First update the position of the views based on the placement info from GL-Native
+        // Then hide the views which are off screen
+        var visibleAnnotationIds: Set<String> = []
+
+        for position in positions {
+            validateAnnotation(byAnnotationId: position.identifier)
+            guard let annotationView = annotationViewsById[position.identifier] else {
+                continue
+            }
+            annotationView.translatesAutoresizingMaskIntoConstraints = true
+            annotationView.frame = CGRect(
+                origin: position.leftTopCoordinate.point,
+                size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
```

Comment:
We have a convenience constructor that converts a Size to a CGSize

Reply:
See Size.swift

Reply:
You mean for the `position`? It's not a `Size` instance, but a custom type that we get for these layout updates.

Reply:
ah, I see